api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -223,6 +224,121 @@ public final Scope startScopedSpan() {
     return CurrentSpanUtils.withSpan(startSpan(), true);
   }
 
+  /**
+   * Wraps a {@link Runnable} so that it executes with the {@code Span} as the current {@code Span},
+   * and ends the {@code Span} after the {@code Runnable} is run. See more details in {@link
+   * Tracer#wrap(Span, Runnable)}.
+   *
+   * <p>Any error will end up as a {@link Status#UNKNOWN}.
+   *
+   * <p>This is a shorter version of the following code:
+   *
+   * <pre><code>
+   * Span span = tracer.spanBuilder("MyRunnableSpan").startSpan();
+   * tracer.wrap(span, new Runnable() {
+   *    {@literal @}Override
+   *     public void run() {
+   *       try {
+   *         myRunnable.run();
+   *       } finally {
+   *         span.end();
+   *       }
+   *     }
+   * });
+   * </code></pre>
+   *
+   * @param runnable the {@code Runnable} to run in the {@code Span}.
+   */
+  public final Runnable startSpanAndWrap(final Runnable runnable) {
+    final Span span = startSpan();
+    return CurrentSpanUtils.wrap(span, runnable, true);
+  }
+
+  /**
+   * Wraps a {@link Callable} so that it executes with the {@code Span} as the current {@code Span},
+   * and ends the {@code Span} after the {@code Callable} is run. See more details in {@link
+   * Tracer#wrap(Span, Callable)}.
+   *
+   * <p>Any error will end up as a {@link Status#UNKNOWN}.
+   *
+   * <p>This is a shorter version of the following code:
+   *
+   * <pre><code>
+   * Span span = tracer.spanBuilder("MyCallableSpan").startSpan();
+   * tracer.wrap(span, {@code new Callable<V>()} {
```

Comment:
Suggestion: The Callable examples would be simpler if they used concrete types instead of `V`.